go/internal/feast/onlineserving/serving.go

```diff
@@ -198,78 +192,86 @@ existed in the registry
 func GetFeatureViewsToUseByFeatureRefs(
 	features []string,
 	registry *registry.Registry,
-	projectName string) ([]*FeatureViewAndRefs, []*SortedFeatureViewAndRefs, []*model.OnDemandFeatureView, error) {
-	viewNameToViewAndRefs := make(map[string]*FeatureViewAndRefs)
-	viewNameToSortedViewAndRefs := make(map[string]*SortedFeatureViewAndRefs)
-	odFvToFeatures := make(map[string][]string)
-	odFvToProjectWithFeatures := make(map[string]*model.OnDemandFeatureView)
+	projectName string) ([]*FeatureViewAndRefs, []*model.OnDemandFeatureView, error) {
 
-	viewToFeaturesValidationMap := make(map[string]map[string]bool)
-	invalidFeatures := make([]string, 0)
+	viewNameToFeatureNames := make(map[string][]string)
 	for _, featureRef := range features {
 		featureViewName, featureName, err := ParseFeatureReference(featureRef)
 		if err != nil {
-			return nil, nil, nil, err
+			return nil, nil, err
+		}
+		if _, exists := viewNameToFeatureNames[featureViewName]; !exists {
+			viewNameToFeatureNames[featureViewName] = make([]string, 0)
```

Comment:
We can define the slice with the first element instead of creating the empty slice and appending.

```suggestion
			viewNameToFeatureNames[featureViewName] = []string{featureName}
```